File: szavazas-core/src/main/java/hu/kdea/szavazas/ballotprocessor/projection/RowProjectionComputer.java
package hu.kdea.szavazas.ballotprocessor.projection;

import boofcv.struct.image.GrayU8;

public final class RowProjectionComputer {
    private RowProjectionComputer() {
    }

    public static float[] compute(GrayU8 gray) {
        float[] projection = new float[gray.height];
        for (int y = 0; y < gray.height; y++) {
            int sum = 0;
            for (int x = 0; x < gray.width; x++) {
                sum += gray.get(x, y);
            }
            projection[y] = (float) sum;
        }
        return projection;
    }
}

File: test-utils/src/main/java/hu/kdea/szavazas/ballotprocessor/projection/RowProjectionComputer.kt
package hu.kdea.szavazas.ballotprocessor

import boofcv.struct.image.GrayU8

object RowProjectionComputer {
    fun compute(gray: GrayU8): FloatArray {
        return FloatArray(gray.height) { y ->
            var sum = 0
            for (x in 0 until gray.width) sum += gray.get(x, y)
            sum.toFloat()
        }
    }
}